airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/source.py

```diff
@@ -65,13 +65,14 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
                 # test that matching on the pattern doesn't error
                 globmatch(file_info.key, config.get("path_pattern"), flags=GLOBSTAR | SPLIT)
                 # just need first file here to test connection and valid patterns
-                return True, None
-
+                break
+            for slice_ in stream.stream_slices(sync_mode=SyncMode.full_refresh):
```

Comment:
added a test read to fail early

Reply:
Could we try to use a custom [AvailabilityStrategy](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/availability_strategy.py#L16)?

Reply:
Do you mind we leave this for a separate PR? I can create an issue for that if we don't have one already. Changes in this PR seem to be quite critical as for me so I'd like to isolate it from other possible issues

Reply:
https://github.com/airbytehq/airbyte/issues/23385

Reply:
https://github.com/airbytehq/airbyte/issues/23385